rocks-db/src/clients/asset_client.rs

```diff
@@ -314,19 +324,129 @@ impl Storage {
+                edition: pagination
+                    .before
+                    .unwrap()
+                    .parse::<u64>()
+                    .expect("failed to parse edition before key"),
+            });
+            revert = true;
+
+            // iterator is on the item we were searching for
+            iter.next();
+            iter
+        } else {
+            let page = pagination.page.unwrap_or(1) - 1;
+            let mut iter = self.db.prefix_iterator_cf(
+                &self.db.cf_handle(TokenMetadataEditionParentIndex::NAME).unwrap(),
+                master_edition.as_ref(),
+            );
+
+            if page > 0 {
+                let skip = page as usize * limit;
+                // iter.skip(n) changes the iterator type, making it incompatible with RocksDB.
+                for _ in 0..skip {
+                    iter.next();
+                }
+            }
+
+            iter
+        };
+
+        let mut asset_keys: Vec<TokenMetadataEditionParentIndex> = Vec::with_capacity(limit);
+        for item in iter {
+            let (_, value) = item?;
+            if let Ok(decoded_value) = TokenMetadataEditionParentIndex::decode(&value) {
+                asset_keys.push(decoded_value);
+            }
+            if asset_keys.len() >= limit {
+                break;
+            }
+        }
+
+        if revert {
+            asset_keys.reverse()
+        }
+
+        Ok(asset_keys)
+    }
+
+    pub async fn get_master_edition_child_assets_info(
+        &self,
+        master_edition: Pubkey,
+        pagination: PaginationQuery,
+    ) -> Result<MasterAssetEditionsInfo> {
+        let (master_edition_metadata, asset_edition_child_assets) = tokio::join!(
+            self.token_metadata_edition_cbor.get_async(master_edition),
+            self.get_master_edition_child_assets(master_edition, pagination),
+        );
+
+        let master_edition_metadata = match master_edition_metadata? {
+            Some(TokenMetadataEdition::MasterEdition(metadata)) => metadata,
+            _ => return Err(StorageError::Common("Expected MasterEdition".to_string())),
+        };
+
+        let asset_edition_info_list: Vec<AssetEditionInfo> = asset_edition_child_assets?
+            .iter()
+            .map(|asset_edition| AssetEditionInfo {
+                mint: asset_edition.asset_key,
+                edition_address: Storage::find_edition_address(&asset_edition.asset_key),
+                edition: asset_edition.edition,
+            })
+            .collect();
+
+        Ok(MasterAssetEditionsInfo {
+            master_edition_address: Storage::find_edition_address(&master_edition),
+            supply: master_edition_metadata.supply,
+            max_supply: master_edition_metadata.max_supply,
+            editions: asset_edition_info_list,
+        })
+    }
+
+    pub fn find_edition_address(mint: &Pubkey) -> Pubkey {
+        let (edition_address, _) = Pubkey::find_program_address(
```

Comment:
As long as you do not need seeds, [create\_program\_address](https://docs.rs/solana-sdk/latest/solana_sdk/pubkey/struct.Pubkey.html#method.create_program_address) is the better option because it's more resource effective.

Reply:
may break if the key matches a regular Pubkey. 
Therefore, it requires additional checks, I am not sure that it will be possible to somehow save resources